nano/node/lmdb/lmdb.cpp

```diff
@@ -687,6 +687,71 @@ void nano::mdb_store::upgrade_v15_to_v16 (nano::write_transaction & transaction_
 		representation = 0;
 	}
 
+	auto account_info_i = latest_begin (transaction_a);
+	auto account_info_n = latest_end ();
+
+	// Set the confirmed frontier for each account in the confirmation height table
+	std::vector<std::pair<nano::account, nano::confirmation_height_info>> confirmation_height_infos;
+	for (nano::mdb_iterator<nano::account, uint64_t> i (transaction_a, confirmation_height), n (nano::mdb_iterator<nano::account, uint64_t>{}); i != n; ++i, ++account_info_i)
```

Comment:
As this upgrade takes a few minutes, maybe do start/stop and periodic logging like is done in `upgrade_v14_to_v15`

Reply:
Good point, I have done that and updated the description of what logs I got from an old upgrade (1.1M accounts) from a v14 ledger. I noticed the vacuum log message doesn't carry forward well with future upgrades, so have updated that too. Also updated the test to make sure it hits all control paths as per private discussion.